added a print on the routing table:
```
--- FAIL: TestPrivilegedReplaceRoute (1.01s)
before: routes found for lo: []
after v4: routes found for lo: []
    route_linux_test.go:98: routes and addresses.
        []
        [127.0.0.1/8 lo ::1/128]
    route_linux_test.go:100: 
        	Error Trace:	/host/pkg/datapath/linux/route/route_linux_test.go:100
        	            				/host/pkg/datapath/linux/route/route_linux_test.go:133
        	Error:      	Received unexpected error:
        	            	no route to host
        	Test:       	TestPrivilegedReplaceRoute
        	Messages:   	for route {Prefix:{IP:f00d::a02:200:0:0 Mask:ffffffffffffffffffffffff00000000} Nexthop:f00d::a02:100:0:815b Local:<nil> Device:lo MTU:0 Priority:0 Proto:0 Scope:universe Table:0 Type:0}
```

looks like the complaint is about not having a local route to the next hop. https://github.com/cilium/cilium/commit/595e7ffd89750431131f002d1b17e5010139c237 seems to fix it

--- a/pkg/datapath/linux/route/route_linux_test.go
+++ b/pkg/datapath/linux/route/route_linux_test.go
@@ -68,7 +68,6 @@ func testReplaceRoute(t *testing.T, prefixStr, nexthopStr string, lookupTest boo
 
 	nexthop := net.ParseIP(nexthopStr)
 	require.NotNil(t, nexthop)
-
 	rt := Route{
 		Device:  "lo",
 		Prefix:  *prefix,
@@ -106,8 +105,24 @@ func testReplaceRoute(t *testing.T, prefixStr, nexthopStr string, lookupTest boo
 func TestPrivilegedReplaceRoute(t *testing.T) {
 	setup(t)
 
+	link, err := safenetlink.LinkByName("lo")
+	require.NoError(t, err)
+
+	// Add addresses to lo within the same subnets as the nexthops so the
+	// kernel accepts the link-scope nexthop routes. The addresses must be
+	// different from the nexthops themselves but in the same subnet.
+	v4Addr, err := netlink.ParseAddr("1.2.3.1/24")
+	require.NoError(t, err)
+	require.NoError(t, netlink.AddrAdd(link, v4Addr))
+	defer netlink.AddrDel(link, v4Addr)
+
+	v6Addr, err := netlink.ParseAddr("f00d::a02:100:0:1/96")
+	require.NoError(t, err)
+	require.NoError(t, netlink.AddrAdd(link, v6Addr))
+	defer netlink.AddrDel(link, v6Addr)
+
 	testReplaceRoute(t, "2.2.0.0/16", "1.2.3.4", true)
-	// lookup test broken for IPv6 as long as use lo as device
+	// lookup test broken for IPv6 as long as we use lo as device
 	testReplaceRoute(t, "f00d::a02:200:0:0/96", "f00d::a02:100:0:815b", false)
 }
 

